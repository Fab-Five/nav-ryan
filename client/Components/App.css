body {
  font-family: "Roboto", Helvetica, sans-serif;
}

#modal-root {
  position: relative;
  z-index: 1000;
}
