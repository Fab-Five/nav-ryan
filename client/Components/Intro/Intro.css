.intro-app {
  margin-top: 61px;
  display: block;
  width: 75%;
  min-width: 75%;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
}

.intro-img-container {
  height: 170px;
  background-size: cover;
  background-position: center;    
  display: flex;
  flex-wrap: wrap;
}

.intro-icon-container {
  flex-basis: 0;
  flex: 1;
  max-width: 100%;
  padding: 12px;
  margin-right: 60px;
}

.intro-back-btn {
  color: #0070eb;
  background: white;
  height: 36px;
  width: 36px;
  border: 0;
  padding: 4px 3px 0 0;
  border-radius: 50%;
  font-size: 1.5em;
  margin-left: 60px;
}

.intro-save-btn {
  color: rgba(0, 0, 0, 0.35);
  background: white;
  height: 36px;
  width: 36px;
  border: 0;
  padding: 4px 0 0 0;
  border-radius: 50%;
  font-size: 1.2em;
  float: right;
  margin-left: 15px;
}

.intro-share-btn {
  color: #0070eb;
  background: white;
  height: 36px;
  width: 36px;
  border: 0;
  padding: 4px 3px 0 0;
  border-radius: 50%;
  font-size: 1.5em;
  float: right;
  margin-left: 15px;
}

.intro-back-btn:hover {
  color: #0058b8;
  cursor: pointer;
}

.intro-save-btn:hover {
  cursor: pointer;
}

.intro-share-btn:hover {
  color: #0058b8;
  cursor: pointer;
}

.intro-summary {
  padding: 12px;
  background-color: white;
}

.intro-summary h1 {
  margin: 0 0 10px 60px;
  font-size: 1.7em;
}

.intro-summary p {
  margin: 0 0 0 60px;
  font-size: 0.9em;
  font-weight: 300;
  color: rgba(0, 0, 0, 0.87);
}

.intro-logo-container {    
  margin-top: -46px;
  margin-right: 60px;
  box-shadow: 0 1px 0 rgba(0, 0, 0, 0.1), 0 0 5px rgba(0, 0, 0, 0.05);
  width: 80px;
  height: 80px;
  position: relative;
  float: right;
}

.intro-logo-container img {
  width: 80px;
  border-radius: 3px;
}

.intro-add {
  margin-right: 16px;
}

.intro-stars {
  width: 120px;
  margin: 10px 5px 0 0;
  cursor: pointer;
}

.intro-num-ratings {
  font-size: 0.8em;
  color: rgba(0, 0, 0, 0.55);
  margin-right: 5px;
  cursor: pointer;
}

.intro-rating-breakdown {
  font-size: 0.8em;
  font-weight: 300;
  color: rgba(0, 0, 0, 0.87);
}

.intro-percent {
  font-size: 1.2em;
  margin-left: 15px;
}

/* tooltip css */
.intro-tooltip {
  display:inline-block;
  position:relative;
  text-align:center;
  margin-left: 60px;
}

.intro-tooltip .intro-tooltip-bottom {
  min-width: 300px; 
  top: 40px;
  left: 110px;
  transform: translate(-40%, 0);
  color: rgba(0, 0, 0, 0.87);
  background-color:#FFFFFF;
  font-size: 14px;
  border: 1px solid rgba(0, 0, 0, 0.1);
  border-radius: 3px;
  position: absolute;
  z-index: 99999999;
  box-sizing: border-box;
  box-shadow: 0 3px 9px rgba(0, 0, 0, 0.5);
  visibility: hidden; 
  opacity: 0; 
  transition: 0.25s ease-out;
}

.intro-tooltip:hover .intro-tooltip-bottom {
  visibility: visible; 
  opacity: 1;
}

.intro-tooltip .intro-tooltip-bottom i {
  position: absolute;
  bottom: 100%;
  left: 40%;
  margin-left: -12px;
  width: 24px;
  height: 12px;
  overflow: hidden;
}

.intro-tooltip .intro-tooltip-bottom i::after {
  content: '';
  position: absolute;
  width: 12px;
  height: 12px;
  left: 50%;
  transform: translate(-50%,50%) rotate(45deg);
  background-color: #FFFFFF;
  box-shadow: 0 1px 8px rgba(0, 0, 0, 0.1);
}

.intro-tooltip-container {
  padding: 20px 10px 10px 10px;
}

.intro-tooltip-header {
  font-size: 1.1em;
  font-weight: 300;
  display: inline-block;
  margin-bottom: 5px;
}

.intro-tooltip-percent {
  font-size: 1.2em;
}

.intro-tooltip-label {
  color: rgba(0, 0, 0, 0.55);
  font-size: 0.9em;
  font-weight: 300;
  line-height: 1.6em;
  display: inline-block;
  margin-top: 5px;
}

.intro-tooltip-breakdown {
  margin-top: 15px;
  width: 32%;
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  box-sizing: border-box;
}

.intro-tooltip-breakdown-mid {
  margin-top: 15px;
  width: 34%;
  padding-left: 10px;
  padding-right: 10px;
  display: inline-block;
  border-left: 1px solid rgba(0, 0, 0, 0.4);
  border-right: 1px solid rgba(0, 0, 0, 0.4);
  box-sizing: border-box;
}

.intro-tooltip-review-container {
  padding-bottom: 30px;
  display: flex;
  flex-direction: row;
  flex-flow: row;
  width: 100%;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
}

.intro-tooltip-avatar {
  border-radius: 50%;
  color: #fff;
  background: #40c2de;
  width: 40px;
  height: 40px;
  border-color: #fff;
  line-height: 40px;
  font-weight: 700;
  margin: 5px 10px 0 15px;
}

.intro-tooltip-review {
  flex: 1 0 0%;
  text-align: left;
  margin-right: 5px;
}

.intro-tooltip-review span:first-of-type {
  display: inline-block;
  margin-bottom: 3px;
}

.intro-tooltip-review span {
  line-height: 1.5em;
  font-weight: 300;
}

.intro-tooltip-all-reviews {
  color: #0070eb;
  padding: 16px;
  font-weight: 300;
  cursor: pointer;
}