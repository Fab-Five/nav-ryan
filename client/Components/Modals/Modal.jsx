import React from 'react';
import ReactDOM from 'react-dom';
// const modalRoot = document.getElementById('modal-root')

class SigninModal extends React.Component {

  constructor(props) {
    super(props);
    this.el = document.createElement('div');
  }

  componentDidMount() {
    document.getElementById('modal-root').appendChild(this.el);
  }

  componentWillUnmount() {
    document.getElementById('modal-root').removeChild(this.el);
  }

  render() {
    return ReactDOM.createPortal(
      this.props.children,
      this.el
    );
  }

}

export default SigninModal;