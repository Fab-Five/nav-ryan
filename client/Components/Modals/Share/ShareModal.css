.share-modal-container {
  background-color: rgba(0,0,0,0.5);
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  animation: 175ms flyUp ease forwards;
}

.share-modal-close {
  font-size: 1.8em;
  color: #0070eb;
}

.share-modal-close:hover {
  cursor: pointer;
  color: #0058b8;
}

.share-modal {
  background-color: white;
  border-radius: 4px;
  width: 45%;
  min-width: 476px; 
  height: 50vh;
  margin-bottom: 10px;
  padding: 2%;
  box-sizing: border-box;
}