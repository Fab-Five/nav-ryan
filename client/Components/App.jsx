import React from 'react';
import Intro from './Intro/Intro.jsx';
import Navbar from './Navbar/Navbar.jsx';
import About from './About/About.jsx';
import axios from 'axios';
import './App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
      <div>
        <Navbar />
        <Intro />
        <About />
      </div>
    );
  }
}

export default App;