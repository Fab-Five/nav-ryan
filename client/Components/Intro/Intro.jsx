import React from 'react';
import axios from 'axios';
import { FaAngleLeft } from 'react-icons/fa';
import { FaShareAlt } from 'react-icons/fa';
import { FaBookmark } from 'react-icons/fa';
import './Intro.css';

class Intro extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      restaurant: {}
    };
    this.fetchRestaurant = this.fetchRestaurant.bind(this);
  }

  // NOTE: Had to have intro component fetch restaurant instead of receiving prop from App because the image
  // was being fetched asynchronously and wasn't responding to CSS styling.
  fetchRestaurant() {
    let url = window.location.href;
    axios.get(`${url}info`)
    .then(({ data }) => {
      let restaurantInfo = data[0];
      this.setState({
        restaurant: restaurantInfo
      });
    })
    .catch((err) => {
      console.log('Could not fetch restaurant: ', err);
    })
  }

  componentDidMount() {
    this.fetchRestaurant();
  }

  render() {
    let { name, image, address, phone, ratings } = this.state.restaurant;
    let address1 = null;
    let stars = null;
    let starSource = null;
    let numRatings = 0;
    let good = 0;
    let timely = 0;
    let correct = 0;
    if(address && ratings) {
      address1 = address['line1'];
      stars = ratings['stars'];
      numRatings = ratings['number'];
      good = ratings['good'];
      timely = ratings['timely'];
      correct = ratings['correct'];
    } 
    if(stars === 2.5) {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/2halfstars.png";
    } else if(stars === 3) {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/3stars.png";
    } else if(stars === 3.5) {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/3halfstars.png";
    } else if(stars === 4) {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/4stars.png";
    } else if(stars === 4.5) {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/4halfstars.png";
    } else {
      starSource = "https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/5stars.png";
    }
    return(<div className="intro-app">
      <div className="intro-img-container" style={{backgroundImage: `url(${image})`}}>
        <div className="intro-icon-container">         
          <button className="intro-back-btn"><FaAngleLeft /></button>
        </div>
        <div className="intro-icon-container">
          <button className="intro-share-btn"><FaShareAlt /></button>
          <button className="intro-save-btn"><FaBookmark /></button>
        </div>
      </div>
      <div className="intro-summary">
        <div className="intro-logo-container">
          <img src="https://s3-us-west-1.amazonaws.com/kayjayhogan/default-img.png"></img>
        </div>
        <h1>{name}</h1>
        <p><span className="intro-add">{address1}</span><span>{phone}</span></p>
        <img src={starSource} className="intro-stars"></img>
        <span className="intro-num-ratings">{numRatings} ratings</span>
        <span className="intro-rating-breakdown">
          <span className="intro-percent"><strong>{good}</strong>%</span> Food was good 
          <span className="intro-percent"><strong>{timely}</strong>%</span> Delivery was on time 
          <span className="intro-percent"><strong>{correct}</strong>%</span> Order was correct
        </span>
      </div>
    </div>);
  }
}

export default Intro;