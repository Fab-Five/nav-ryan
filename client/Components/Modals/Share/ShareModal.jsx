import React from 'react';
import Modal from '../Modal.jsx';
import { MdClose } from 'react-icons/md';
import { MdEmail } from 'react-icons/md';
import { FaFacebookSquare } from 'react-icons/fa';
import { FaTwitter } from 'react-icons/fa';
import { FaBolt } from 'react-icons/fa';
import './ShareModal.css';

class ShareModal extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      copyLinkLabel: 'Copy link'
    }
    this.handleCopy = this.handleCopy.bind(this);
  }

  handleCopy() {
    this.setState({
      copyLinkLabel:'Link copied'
    });
  }

  render() {
    let copyLinkLabelStyle = null;
    if(this.state.copyLinkLabel === 'Copy link') {
      copyLinkLabelStyle = {color: "#0070eb"}
    } else {
      copyLinkLabelStyle = {color: "#13aa37"}
    }
    return(<Modal>
      <div className="share-modal-container">
        <div className="share-modal">
          <span className="share-modal-close" onClick={this.props.handleHide}><MdClose /></span>
          <h3>Share {this.props.restaurant}</h3>
          <div className="share-options-container">
            <div className="share-option">
              <FaFacebookSquare /> 
              <span className="share-option-label">Facebook</span>
            </div>
            <div className="share-option">
              <FaTwitter /> 
              <span className="share-option-label">Twitter</span>
            </div>
            <div className="share-option">
              <MdEmail /> 
              <span className="share-option-label">Email</span>
            </div>
            <div className="share-option" id="share-copy-link" onClick={this.handleCopy}>
              <FaBolt/> 
              <span style={copyLinkLabelStyle} className="share-option-label">{this.state.copyLinkLabel}</span>
            </div>
          </div>
        </div>
      </div>
    </Modal>);
  }
}

export default ShareModal;