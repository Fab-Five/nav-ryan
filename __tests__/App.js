import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/Components/App';
import Intro from '../client/Components/Intro/Intro';
import Navbar from '../client/Components/Navbar/Navbar';
import Modal from '../client/Components/Modals/Modal';
import ShareModal from '../client/Components/Modals/Share/ShareModal';
import SigninModal from '../client/Components/Modals/Signin/SigninModal';

describe('Testing the App Component', () => {

   it('renders the App component', ()=> {
      expect(shallow(<App />).length).toEqual(1)
   });

   it('should grab a snapshot of the component', ()=> {
		const component = renderer.create(<App />)
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
   });
   
});
