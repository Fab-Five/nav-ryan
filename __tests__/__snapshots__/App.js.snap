// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App Component should grab a snapshot of the component 1`] = `
<div>
  <div
    className="nav-app"
  >
    <div
      className="nav-navbar"
    >
      <div
        className="nav-brand"
      >
        <img
          src="https://s3-us-west-1.amazonaws.com/kayjayhogan/GH.png"
        />
      </div>
      <div
        className="nav-search"
      >
        <div
          className="nav-search-address-container"
        >
          <div
            className="nav-search-address"
            onClick={[Function]}
          >
            <span
              className="nav-loc-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                />
              </svg>
            </span>
            <span
              className="nav-search-address-text"
            >
              6060 Center Dr
            </span>
            <span
              className="nav-down-arrow"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 10 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 11L0 6l1.5-1.5L5 8.25 8.5 4.5 10 6l-5 5z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          className="nav-search-input-container"
        >
          <div
            className="nav-search-flex"
          >
            <span
              className="nav-search-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
            <input
              onClick={[Function]}
              placeholder="Pizza, sushi, chinese"
            />
            <span
              className="nav-close-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        className="nav-buttons"
      >
        <div
          className="nav-sign-in"
        >
          <button
            className="nav-sign-in-btn"
            onClick={[Function]}
          >
            <strong>
              Sign In
            </strong>
          </button>
        </div>
        <div
          className="nav-bag"
        >
          <span
            className="nav-bag-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="intro-app"
  >
    <div
      className="intro-img-container"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    >
      <div
        className="intro-icon-container"
      >
        <button
          className="intro-back-btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 256 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
            />
          </svg>
        </button>
      </div>
      <div
        className="intro-icon-container"
      >
        <button
          className="intro-share-btn"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
            />
          </svg>
        </button>
        <button
          className="intro-save-btn"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="intro-summary"
    >
      <div
        className="intro-logo-container"
      >
        <img
          src="https://s3-us-west-1.amazonaws.com/kayjayhogan/default-img.png"
        />
      </div>
      <h1 />
      <p>
        <span
          className="intro-add"
        />
        <span />
      </p>
      <div
        className="intro-tooltip"
      >
        <img
          className="intro-stars"
          src="https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/5stars.png"
        />
        <span
          className="intro-num-ratings"
        >
          0
           ratings
        </span>
        <div
          className="intro-tooltip-bottom"
        >
          <div
            className="intro-tooltip-container"
          >
            <span
              className="intro-tooltip-header"
            >
              Here's what people are saying:
            </span>
            <div
              className="intro-tooltip-breakdown"
            >
              <span
                className="intro-tooltip-percent"
              >
                <strong>
                  0
                </strong>
                %
              </span>
              <br />
              <span
                className="intro-tooltip-label"
              >
                Food was good
              </span>
            </div>
            <div
              className="intro-tooltip-breakdown-mid"
            >
              <span
                className="intro-tooltip-percent"
              >
                <strong>
                  0
                </strong>
                %
              </span>
              <br />
              <span
                className="intro-tooltip-label"
              >
                Delivery was on time
              </span>
            </div>
            <div
              className="intro-tooltip-breakdown"
            >
              <span
                className="intro-tooltip-percent"
              >
                <strong>
                  0
                </strong>
                %
              </span>
              <br />
              <span
                className="intro-tooltip-label"
              >
                Order was correct
              </span>
            </div>
          </div>
          <div
            className="intro-tooltip-review-container"
          >
            <div
              className="intro-tooltip-avatar"
            />
            <div
              className="intro-tooltip-review"
            >
              <span>
                <strong>
                  :
                </strong>
              </span>
              <br />
              <span />
            </div>
          </div>
          <div
            className="intro-tooltip-all-reviews"
          >
            All Reviews (
            0
            )
          </div>
          <i />
        </div>
      </div>
      <span
        className="intro-rating-breakdown"
      >
        <span
          className="intro-percent"
        >
          <strong>
            0
          </strong>
          %
        </span>
         Food was good
        <span
          className="intro-percent"
        >
          <strong>
            0
          </strong>
          %
        </span>
         Delivery was on time
        <span
          className="intro-percent"
        >
          <strong>
            0
          </strong>
          %
        </span>
         Order was correct
      </span>
    </div>
  </div>
</div>
`;

exports[`Testing the Intro Component should grab a snapshot of the component 1`] = `
<div
  className="intro-app"
>
  <div
    className="intro-img-container"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  >
    <div
      className="intro-icon-container"
    >
      <button
        className="intro-back-btn"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 256 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          />
        </svg>
      </button>
    </div>
    <div
      className="intro-icon-container"
    >
      <button
        className="intro-share-btn"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352 320c-22.608 0-43.387 7.819-59.79 20.895l-102.486-64.054a96.551 96.551 0 0 0 0-41.683l102.486-64.054C308.613 184.181 329.392 192 352 192c53.019 0 96-42.981 96-96S405.019 0 352 0s-96 42.981-96 96c0 7.158.79 14.13 2.276 20.841L155.79 180.895C139.387 167.819 118.608 160 96 160c-53.019 0-96 42.981-96 96s42.981 96 96 96c22.608 0 43.387-7.819 59.79-20.895l102.486 64.054A96.301 96.301 0 0 0 256 416c0 53.019 42.981 96 96 96s96-42.981 96-96-42.981-96-96-96z"
          />
        </svg>
      </button>
      <button
        className="intro-save-btn"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 512V48C0 21.49 21.49 0 48 0h288c26.51 0 48 21.49 48 48v464L192 400 0 512z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="intro-summary"
  >
    <div
      className="intro-logo-container"
    >
      <img
        src="https://s3-us-west-1.amazonaws.com/kayjayhogan/default-img.png"
      />
    </div>
    <h1 />
    <p>
      <span
        className="intro-add"
      />
      <span />
    </p>
    <div
      className="intro-tooltip"
    >
      <img
        className="intro-stars"
        src="https://s3-us-west-1.amazonaws.com/kayjayhogan/Stars/5stars.png"
      />
      <span
        className="intro-num-ratings"
      >
        0
         ratings
      </span>
      <div
        className="intro-tooltip-bottom"
      >
        <div
          className="intro-tooltip-container"
        >
          <span
            className="intro-tooltip-header"
          >
            Here's what people are saying:
          </span>
          <div
            className="intro-tooltip-breakdown"
          >
            <span
              className="intro-tooltip-percent"
            >
              <strong>
                0
              </strong>
              %
            </span>
            <br />
            <span
              className="intro-tooltip-label"
            >
              Food was good
            </span>
          </div>
          <div
            className="intro-tooltip-breakdown-mid"
          >
            <span
              className="intro-tooltip-percent"
            >
              <strong>
                0
              </strong>
              %
            </span>
            <br />
            <span
              className="intro-tooltip-label"
            >
              Delivery was on time
            </span>
          </div>
          <div
            className="intro-tooltip-breakdown"
          >
            <span
              className="intro-tooltip-percent"
            >
              <strong>
                0
              </strong>
              %
            </span>
            <br />
            <span
              className="intro-tooltip-label"
            >
              Order was correct
            </span>
          </div>
        </div>
        <div
          className="intro-tooltip-review-container"
        >
          <div
            className="intro-tooltip-avatar"
          />
          <div
            className="intro-tooltip-review"
          >
            <span>
              <strong>
                :
              </strong>
            </span>
            <br />
            <span />
          </div>
        </div>
        <div
          className="intro-tooltip-all-reviews"
        >
          All Reviews (
          0
          )
        </div>
        <i />
      </div>
    </div>
    <span
      className="intro-rating-breakdown"
    >
      <span
        className="intro-percent"
      >
        <strong>
          0
        </strong>
        %
      </span>
       Food was good
      <span
        className="intro-percent"
      >
        <strong>
          0
        </strong>
        %
      </span>
       Delivery was on time
      <span
        className="intro-percent"
      >
        <strong>
          0
        </strong>
        %
      </span>
       Order was correct
    </span>
  </div>
</div>
`;

exports[`Testing the Navbar Component should grab a snapshot of the component 1`] = `
<div
  className="nav-app"
>
  <div
    className="nav-navbar"
  >
    <div
      className="nav-brand"
    >
      <img
        src="https://s3-us-west-1.amazonaws.com/kayjayhogan/GH.png"
      />
    </div>
    <div
      className="nav-search"
    >
      <div
        className="nav-search-address-container"
      >
        <div
          className="nav-search-address"
          onClick={[Function]}
        >
          <span
            className="nav-loc-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
          </span>
          <span
            className="nav-search-address-text"
          >
            6060 Center Dr
          </span>
          <span
            className="nav-down-arrow"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 10 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 11L0 6l1.5-1.5L5 8.25 8.5 4.5 10 6l-5 5z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="nav-search-input-container"
      >
        <div
          className="nav-search-flex"
        >
          <span
            className="nav-search-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
                fillRule="evenodd"
              />
            </svg>
          </span>
          <input
            onClick={[Function]}
            placeholder="Pizza, sushi, chinese"
          />
          <span
            className="nav-close-icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      className="nav-buttons"
    >
      <div
        className="nav-sign-in"
      >
        <button
          className="nav-sign-in-btn"
          onClick={[Function]}
        >
          <strong>
            Sign In
          </strong>
        </button>
      </div>
      <div
        className="nav-bag"
      >
        <span
          className="nav-bag-icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
